bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/internal/Ism8Handler.java

```diff
@@ -0,0 +1,158 @@
+        }
+    }
+
+    @Override
+    public void initialize() {
+        this.config = getConfigAs(Ism8Configuration.class);
+        try {
+            Ism8Configuration cfg = this.config;
+            if (cfg != null) {
+                Server svr = new Server(cfg.getPortNumber());
+                for (Channel channel : getThing().getChannels()) {
+                    if (channel.getConfiguration().containsKey("id")
+                            && channel.getConfiguration().containsKey("type")) {
+                        int id = Integer.parseInt(channel.getConfiguration().get("id").toString());
+                        String type = channel.getConfiguration().get("type").toString();
+                        String description = channel.getLabel();
+                        if (type != null && description != null) {
+                            svr.addDataPoint(id, type, description);
+                        }
+                    } else {
+                        this.logger.warn("Ism8: ID or type missing - Channel={}  Cfg={}", channel.getLabel(),
+                                channel.getConfiguration());
+                    }
+                    this.logger.debug("Ism8: Channel={}", channel.getConfiguration().toString());
+                }
+
+                this.updateStatus(ThingStatus.UNKNOWN);
+                svr.addDataPointChangeListener(this);
+                scheduler.execute(svr::start);
+                this.server = svr;
+            }
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            this.logger.warn("Ism8 Initialize: {}", e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void dataPointChanged(@Nullable DataPointChangedEvent e) {
+        if (e != null) {
+            IDataPoint dataPoint = e.getDataPoint();
+            if (dataPoint != null) {
+                this.logger.debug("Ism8: dataPointChanged {}", dataPoint.toString());
+                this.updateDataPoint(dataPoint);
+            }
+        }
+    }
+
+    @Override
+    public void connectionStatusChanged(ThingStatus status) {
+        this.updateStatus(status);
+    }
+
+    private void updateDataPoint(IDataPoint dataPoint) {
+        this.updateStatus(ThingStatus.ONLINE);
+        for (Channel channel : getThing().getChannels()) {
+            try {
+                if (channel.getConfiguration().containsKey("id")
+                        && Integer.parseInt(channel.getConfiguration().get("id").toString()) == dataPoint.getId()) {
+                    logger.debug("Ism8 updateDataPoint ID:{} {}", dataPoint.getId(), dataPoint.getValueText());
+                    Object val = dataPoint.getValueObject();
+                    if (val != null) {
+                        updateState(channel.getUID(), new QuantityType<>(val.toString()));
+                    }
+                }
+            } catch (Exception e) {
```

Comment:
You should specify the concrete exception you want to catch, since catching `Exception` catches also all runtime exceptions.

Reply:
I was basically catching the Exception by default, not checking the potential cases. In this case it might be a wrong configuration ID that might not be convertable into an integer. I changed the Exception to the NumberFormatException.